### PR TITLE
[stdlib] Improve readability of `test_python_object`

### PR DESCRIPTION
**Pull request by jjvraw**
Follow up on, 
- #3549 

CC @JoeLoser

### CODE REVIEW

**Comment by jjvraw** [comment]
Marked as draft, until #3549 is merged.

**Comment by JoeLoser** [comment]
!sync

File: stdlib/test/python/test_python_object.mojo [hunk 1]
```diff
@@ -502,21 +485,95 @@ def test_setitem_raises():
...
+
+    var empty = PythonObject([])
+    assert_equal("[]", str(empty[:]))
+    assert_equal("[]", str(empty[1:2:3]))
+
+    var d = Python.dict()
+    d["a"] = 1
+    d["b"] = 2
+    with assert_raises(contains="slice(1, 3, None)"):
+        _ = d[1:3]
```
Comment (jjvraw) [inline stdlib/test/python/test_python_object.mojo:544]: Note, this is still included. 

Unsure how it was handled internally.

**Comment by JoeLoser** [comment]
!sync

**Pull request closed without merging**